src_python/habitat_sim/simulator.py

```diff
@@ -646,72 +649,85 @@ def __init__(self, sim: Simulator, agent: Agent, sensor_id: str) -> None:
 
     def draw_observation(self) -> None:
-        assert self._sim.renderer is not None
-        # see if the sensor is attached to a scene graph, otherwise it is invalid,
-        # and cannot make any observation
-        if not self._sensor_object.object:
-            raise habitat_sim.errors.InvalidAttachedObject(
-                "Sensor observation requested but sensor is invalid.\
-                 (has it been detached from a scene node?)"
-            )
-        self._sim.renderer.draw(self._sensor_object, self._sim)
+        if self._spec.sensor_type == SensorType.AUDIO:
+            # do nothing in draw observation, get_observation will be called after this
+            # run the simulation there
+            return
+        else:
+            assert self._sim.renderer is not None
+            # see if the sensor is attached to a scene graph, otherwise it is invalid,
+            # and cannot make any observation
+            if not self._sensor_object.object:
+                raise habitat_sim.errors.InvalidAttachedObject(
+                    "Sensor observation requested but sensor is invalid.\
+                    (has it been detached from a scene node?)"
+                )
+            self._sim.renderer.draw(self._sensor_object, self._sim)
 
     def _draw_observation_async(self) -> None:
-        assert self._sim.renderer is not None
-        if (
-            self._spec.sensor_type == SensorType.SEMANTIC
-            and self._sim.get_active_scene_graph()
-            is not self._sim.get_active_semantic_scene_graph()
-        ):
-            raise RuntimeError(
-                "Async drawing doesn't support semantic rendering when there are multiple scene graphs"
-            )
-        # TODO: sync this path with renderer changes as above (render from sensor object)
-
-        # see if the sensor is attached to a scene graph, otherwise it is invalid,
-        # and cannot make any observation
-        if not self._sensor_object.object:
-            raise habitat_sim.errors.InvalidAttachedObject(
-                "Sensor observation requested but sensor is invalid.\
-                 (has it been detached from a scene node?)"
-            )
-
-        # get the correct scene graph based on application
-        if self._spec.sensor_type == SensorType.SEMANTIC:
-            if self._sim.semantic_scene is None:
+        if self._spec.sensor_type == SensorType.AUDIO:
+            # do nothing in draw observation, get_observation will be called after this
+            # run the simulation there
+            return
+        else:
+            assert self._sim.renderer is not None
+            if (
+                self._spec.sensor_type == SensorType.SEMANTIC
+                and self._sim.get_active_scene_graph()
+                is not self._sim.get_active_semantic_scene_graph()
+            ):
                 raise RuntimeError(
                     "SemanticSensor observation requested but no SemanticScene is loaded"
```

Comment:
Looks like a merge issue here.
```suggestion
                    "Async drawing doesn't support semantic rendering when there are multiple scene graphs"
```